Bug 1808358 - Use OnboardingPage in UpgradeOnboarding

rahulsainani opened the pull request:
### What
– Rename to UpgradeOnboarding
– Use the common OnboardingPage component in UpgradeOnboarding.
– No visual changes except that the indicator doesn't scroll as that's not part of the `OnboardingPage`

### Possible improvements
– Create a common Indiciator composable and `IndicatorState` that is part of `OnboardingPage`

### Screenshots
| Welcome | Sync sign in | 
|------|-------|
|<img width="296" alt="Welcome" src="https://user-images.githubusercontent.com/38040960/219050869-f60f3d3b-8541-4bbb-916a-7e5ba2a28cc0.png">|<img width="296" alt="Sync Sign-in" src="https://user-images.githubusercontent.com/38040960/219050898-70518736-4abf-409b-b98d-d8c60abf38e7.png">


### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.







### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1808358

gabrielluong commented:
Was having issues building locally after the monorepo migration - only managed to get my local development on the monorepo going just now. Will look at this first time in the morning

rahulsainani commented:
No worries! thanks for the heads up! 👍

gabrielluong commented on fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/UpgradeOnboarding.kt:
```diff
@@ -62,200 +50,96 @@ private enum class OnboardingState {
-        if (isSyncSignIn) {
-            Spacer(modifier = Modifier.height(6.dp))
-        } else {
+        if (!isSyncSignIn) {
```
Changes look good to me. However, we have removed the following condition:

```
if (isSyncSignIn) {
            Spacer(modifier = Modifier.height(6.dp))
}
```
This simply fills in the 6dp height if the Indicators wasn't visible. While this probably doesn't amount to much in the grand scheme of things, but I rather ensure we are keeping the bottom padding of the Button to the bottom of the screen to still align at a total of 70dp. 

https://www.figma.com/file/d7l1iGabQYqnLcnQbHDGzM/Wallpaper-2.0-%2B-Onboarding-v106?node-id=6058%3A209483&t=4jqlRpeR0Cday80G-0

rahulsainani replied:
good catch! I've updated this 👍

gabrielluong commented on fenix/app/src/main/res/values/strings.xml:
```diff
@@ -274,7 +274,7 @@
     <string name="sync_cfr_message">Your tabs are syncing! Pick up where you left off on your other device.</string>
     <!-- Content description (not visible, for screen readers etc.): Close button for the home onboarding dialog -->
-    <string name="onboarding_home_content_description_close_button">Close</string>
+    <string name="onboarding_home_content_description_close_button" moz:RemovedIn="112" tools:ignore="UnusedResources">Close</string>
```
I now see that we ended up using a 'common' content description for all these Close buttons. In the past, I have attempted to simplify the duplication of the multiple common strings we had such as Ok, Cancel, etc. However, this isn't quite right from a localization point of view. We generally would rather duplicate the string and provide context for each of their different usage. 

This was the most recent discussion because a contributor also wanted to de-duplicate the strings to have a common localized strings. https://github.com/mozilla-mobile/fenix/pull/27813#issuecomment-1316080412

rahulsainani replied:
Ah I see, yeah the intention was to use the common strings for a common component, but seeing it has been an issue and has been discussed, it makes sense to use specific strings. Thanks for sharing the discussions, I've updated it!

rahulsainani commented:
@HarrisonOg @mavduevskiy The [recent RTL fix merge](https://github.com/mozilla-mobile/firefox-android/pull/850) had some conflicts with this PR, so I rebased and updated to include those changes. Tagging you to get some 👀 on it, please flag if you see something is off, thanks!
